synthesizer/src/snark/verifying_key/mod.rs

```diff
@@ -59,7 +59,7 @@ impl<N: Network> VerifyingKey<N> {
     }
 
     /// Returns `true` if the batch proof is valid for the given public inputs.
-    pub fn verify_batch(&self, function_name: &Identifier<N>, inputs: &[&[N::Field]], proof: &Proof<N>) -> bool {
+    pub fn verify_batch(&self, function_name: &str, inputs: &[Vec<N::Field>], proof: &Proof<N>) -> bool {
```

Comment:
what's the rationale behind this change? is this done in order for `inputs` to go out of scope at the callsite faster?

Reply:
I had same remark. I want to see if we can return it to a slice.

Reply:
I reverted the `verify_batch` back to slices. The one difference is that the `state_path_verifier_inputs` now uses the following logic:
`&state_path_verifier_inputs.iter().map(|inputs| inputs.as_slice()).collect::<Vec<_>>()`

Reply:
I reverted that change, I don't think incurring a collect is faster.